controllers/reconcile_operator.go

```diff
@@ -192,7 +192,11 @@ func (r *OperandRequestReconciler) deleteSubscription(operandName string, reques
 		return nil
 	}
 
-	csv, err := fetch.FetchClusterServiceVersion(r.Client, sub)
+	if sub.Status.CurrentCSV == "" {
```

Comment:
If the `CurrentCSV` is empty, which could be caused by some OLM error. Consider this situation, OLM can't generate Installplan or CSV for the operand.

ODLM should be able to delete the subscription as well. Thus I think we can delete the subscription only if the `CurrentCSV` is empty

Reply:
In upgrade process, maybe the CurrentCSV will disappear for a while, if we only delete sub, for the next reconcile, we will not delete the custom resource and CSV